src/test/java/org/opensearch/neuralsearch/processor/NormalizationProcessorIT.java

```diff
@@ -471,6 +473,301 @@ public void testMinMaxLowerBounds_whenLowerBoundsIsGreaterThenActualMinScore_the
         assertQueryResults(searchResponseAsMapPartialMatch, 4, false, Range.between(0.33f, 1.0f));
     }
 
+    @SneakyThrows
+    public void testMinMaxUpperBounds_whenMultipleShards_thenSuccessful() {
```

Comment:
Do we have a test for single shard scenario?

Reply:
Added